start/manage/update.php

```diff
@@ -17,57 +13,117 @@
 
 $ship = $shipLoader->findOneById($id);
 
+
 if ($_SERVER['REQUEST_METHOD'] == 'POST'){
-    header('location:index.php');die;
+
+
+    $ship->setName($_POST['ship']);
+    $ship->setWeaponPower($_POST['weaponPower']);
+    $ship->setJediFactor($_POST['jediFactor']);
+    $ship->setStrength($_POST['strength']);
+    $ship->setDescription($_POST['description']);
+
+
+    if (empty(trim($ship->getName()))){
+        $errmessages[] = "Please enter ship name";
+    }
+
+    if (empty(trim($ship->getWeaponPower()))){
+        $errmessages[] = "Please enter weapon power";
+    } elseif ($ship->getWeaponPower() < 0){
+        $errmessages[] = "Weapon Power can't be less than zero";
+    } elseif (is_int($ship->getWeaponPower() == false)) {
+        $errmessages[] = "Weapn Power must be a number";
+    }
+    
+    if (empty(trim($ship->getJediFactor()))){
+        $errmessages[] = "Please enter Jedi Factor";
+    } elseif ($ship->getJediFactor() < 0){
+        $errmessages[] = "Jedi Factor can't be less than zero";
+    } elseif (is_int($ship->getJediFactor() == false)){
+        $errmessages[] = "Jedi Factor must be a number";
+    }
+
+    if (empty(trim($ship->getStrength()))){
+        $errmessages[] = "Please enter ship strength";
+    } elseif ($ship->getStrength() < 0) {
+        $errmessages[] = "Strength can't be less than zero";
+    } elseif (is_int($ship->getstrength() == false)){
+        $errmessages[] = "Strength must be a number";
+    }
+
+    if (empty(trim($ship->getDescription()))){
+        $errmessages[] = "Please enter ship description";
+    }
+
+    if ($errmessages == []){
+
+        $shipStorage = $container->getShipStorage();
+        $shipStorage->updateShip($ship);
+
+        header('location: index.php');
+    }
+
 }
 
 ?>
 
 <div class='container'>
     <a href="show.php?id=<?php echo $id?>"> Back to previous page </a>
```

Comment:
It might be nice to have a breadcrumb here instead of just going back one page.

It would be something like `Manage ships [link to index] > Current ship name [link to show page] > Edit`